### DISCUSSION TITLE
EfficientAD Epochs and Validation split

### DISCUSSION THREAD

**Discussion by Muhammad-237** [open-edge-platform/anomalib#2774] [Q&A] [answered]
1. i ma training efficientAd model with defaults setting but it is taking long time and i have didn't find anything related to epochs in documentation or in github repo efficientad code class and not even in engine class. is there anyway i can change epochs

My Code:

from anomalib.data import Folder
from anomalib.models import EfficientAd
from anomalib.engine import Engine
from anomalib.data.utils import (
    DirType,
    LabelName,
    Split,
    TestSplitMode,
    ValSplitMode,
)

datamodule = Folder(
    name="hexa_adidas_logo_new",
    root=r"C:\Users\ai1\Desktop\EfficientAD\dataset\hexa_adidas_logo_new",
    normal_dir="good",
    abnormal_dir="abnormal",
    test_split_mode =TestSplitMode.NONE,
    val_split_mode=ValSplitMode.FROM_TRAIN
)

datamodule.setup()

model = EfficientAd()
engine = Engine()

engine.fit(model=model, datamodule=datamodule)

2. Next Question is related to validation split as i have also added abnormal images but still getting this output in code 

Calculate Validation Dataset Quantiles: 100%|██████████| 6/6 [00:08<00:00,  1.37s/it]
WARNING:root:The validation set does not contain any anomalous images. As a result, the adaptive threshold will take the value of the highest anomaly score observed in the normal validation images, which may lead to poor predictions. For a more reliable adaptive threshold computation, please add some anomalous images to the validation set.
WARNING:root:The validation set does not contain any anomalous images. As a result, the adaptive threshold will take the value of the highest anomaly score observed in the normal validation images, which may lead to poor predictions. For a more reliable adaptive threshold computation, please add some anomalous images to the validation set.

**Comment by Muhammad-237** [comment]
still waiting for the response

**Comment by blaz-r** [comment, marked as answer]
Hi, Engine takes almost all arguments that the [lighting trainer ](https://lightning.ai/docs/pytorch/stable/common/trainer.html) does. So you can either set [max_epochs ](https://lightning.ai/docs/pytorch/stable/common/trainer.html#max-epochs) argument or [max_steps](https://lightning.ai/docs/pytorch/stable/common/trainer.html#max-steps).

About abnormal samples, you'll need to add them to validation or test folder, depending on your setup. Since the learning type if unsupervised, they will be ignored in (removed from) training set.

**Comment by github-actions** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.
